xxHash32/xxHash32.go

```diff
@@ -203,3 +211,8 @@ func Checksum(input []byte, seed uint32) uint32 {
 
 	return h32
 }
+
+func u32(buf []byte) uint32 {
+	// go compiler recognizes this pattern and optimizes it on little endian platforms
```

Comment:
Initially I did the same thing using convoluted unsafe.Pointer casts and little endian autp-detection before noticing the compiler is able to detect this pattern.